dashboard/src/components/AppView/AccessURLTable/AccessURLTable.tsx

```diff
@@ -1,49 +1,76 @@
 import * as React from "react";
 
-import { IResource, IServiceSpec } from "../../../shared/types";
+import LoadingWrapper from "../../../components/LoadingWrapper";
+import { IKubeItem, IResource, IServiceSpec } from "../../../shared/types";
+import isSomeResourceLoading from "../helpers";
 import AccessURLItem from "./AccessURLItem";
 import { GetURLItemFromIngress } from "./AccessURLItem/AccessURLIngressHelper";
 import { GetURLItemFromService } from "./AccessURLItem/AccessURLServiceHelper";
 
 interface IServiceTableProps {
-  services: { [s: string]: IResource };
-  ingresses: { [i: string]: IResource };
+  services: Array<IKubeItem<IResource>>;
+  ingresses: Array<IKubeItem<IResource>>;
 }
 
 class AccessURLTable extends React.Component<IServiceTableProps> {
   public render() {
-    const { services, ingresses } = this.props;
+    const { ingresses, services } = this.props;
+    let ingressSection = <p>The current application does not expose a public URL.</p>;
```

Comment:
whilst both services and ingresses are ingresses, the term is confusing because at first I thought we were excluding services here. maybe we should call this accessTable or something else more generic?